pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -270,8 +274,11 @@ func (ca *cloudCIDRAllocator) Run(stopCh <-chan struct{}) {
 	ctx, cancelFn := context.WithCancel(context.Background())
 	defer cancelFn()
 	defer ca.queue.ShutDown()
-	defer ca.nodeTopologyQueue.Shutdown()
-
+	if enableNodeTopology {
```

Comment:
Maybe combine this if with the one below? Having them separately doesn't seem to be functionally different